django/db/backends/base/operations.py

```diff
@@ -714,42 +714,68 @@ def subtract_temporals(self, internal_type, lhs, rhs):
-            % start
-        )
+            else:
+                return "%d %s" % (value, self.FOLLOWING)
+        elif value is None:
+            return none_value
+
+    def window_frame_rows_start(self, start):
+        as_sql = self.window_frame_value(start, none_value=self.UNBOUNDED_PRECEDING)
+        if as_sql is None:
+            raise ValueError(
+                "start argument must be an integer, zero, or None, but got '%s'."
+                % start
+            )
+        return as_sql
 
-    def window_frame_end(self, end):
-        if isinstance(end, int):
-            if end == 0:
-                return self.CURRENT_ROW
-            elif end > 0:
-                return "%d %s" % (end, self.FOLLOWING)
-        elif end is None:
-            return self.UNBOUNDED_FOLLOWING
-        raise ValueError(
-            "end argument must be a positive integer, zero, or None, but got '%s'."
-            % end
-        )
+    def window_frame_rows_end(self, end):
+        as_sql = self.window_frame_value(end, none_value=self.UNBOUNDED_FOLLOWING)
+        if as_sql is None:
+            raise ValueError(
+                "end argument must be an integer, zero, or None, but got '%s'." % end
+            )
+        return as_sql
+
+    def window_frame_range_start(self, start):
+        as_sql = self.window_frame_value(start, none_value=self.UNBOUNDED_PRECEDING)
+        if isinstance(start, int) and start > 0 or as_sql is None:
+            raise ValueError(
+                "start argument must be a negative integer, zero, or None, "
+                "but got '%s'." % start
+            )
+        return as_sql
+
+    def window_frame_range_end(self, end):
+        as_sql = self.window_frame_value(end, none_value=self.UNBOUNDED_FOLLOWING)
+        if isinstance(end, int) and end < 0 or as_sql is None:
+            raise ValueError(
+                "end argument must be a positive integer, zero, or None, but got '%s'."
+                % end
+            )
+        return as_sql
 
     def window_frame_rows_start_end(self, start=None, end=None):
         """
         Return SQL for start and end points in an OVER clause window frame.
         """
         if not self.connection.features.supports_over_clause:
             raise NotSupportedError("This backend does not support window expressions.")
-        return self.window_frame_start(start), self.window_frame_end(end)
+        elif isinstance(start, int) and isinstance(end, int) and start > end:
+            raise ValueError("start cannot be greater than end.")
+        return self.window_frame_rows_start(start), self.window_frame_rows_end(end)
 
     def window_frame_range_start_end(self, start=None, end=None):
-        start_, end_ = self.window_frame_rows_start_end(start, end)
+        if not self.connection.features.supports_over_clause:
+            raise NotSupportedError("This backend does not support window expressions.")
```

Comment:
Do you think it's worth adding? `WindowFrame` subclasses are the only one that call this method and they can only be used with `Window` which already raises an exception when `supports_over_clause` is `False`.

Reply:
Ah clever - you're right, it's not required 👍